### DISCUSSION TITLE
What are benefits of cache settings "Multi-front optimizations"?

### DISCUSSION THREAD

**Discussion by ShaiMagal** [PrestaShop/PrestaShop#37350] [Ideas]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

Please, can someone explain me, what is benefit of using cache settings "Multi-front optimizations"?
What are benefits of this, when all quality hosting companies have already NVMe gen4 (1 000 000 IOPS) drives longer time?

We already removed mysql cache rom PS8, so why we don't remove this option for PS9?

**Is there any real use case, when this settings can be usefull?**

Please, let me know :)

I'm guessing that the end customer doesn't know what it's for anyway and simply "turns everything on" to try make PS go quickly. Maybe it's a good time to discuss and possibly remove this option?

![Untitled](https://github.com/user-attachments/assets/f79dc16e-8122-4151-9896-387c59373bc2)

### Steps to reproduce

Discussion

### PrestaShop version(s) where the bug happened

9.0.0

### PHP version(s) where the bug happened

8.3

### Your company or customer's name goes here (if applicable).

https://www.openservis.cz/

**Comment by matks** [comment]
I think multi front optimization was introduced by @jocel1 a long time ago.

> What are benefits of this, when all quality hosting companies have already NVMe gen4 (1 000 000 IOPS) drives longer time?

AFAIK multi-server infrastructure is still a valid choice for PrestaShop. I don't think mentioning hosting prices is a good reason to add or reject a feature 😉 .

**Comment by jocel1** [comment]
@matks @ShaiMagal actually this feature was introduced initially to try to reduce the cost of the smarty cache when the whole code base is stored on a NFS file system (which is often used with multi-server infra). This option allows to store the smarty cache locally on each servers (could be on a tmpfs for example) thanks to a mecanism to handle their invalidation.
However it introduces overhead on the database side, so it should be used really carefully

**Comment by ShaiMagal** [comment]
@matks I mean this is totally useless feature, never see any benefit of this... So I am asking, if we need it? Now skip hosting prices, but focus on this feature. Is it really usefull? If yes, we should edit "help box" for this funcion to explain, when it can help. Never see any benefit.

@jocel1 I understand, but, we need this? I never see any web benefits from this feature...

@Hlavtox  WDYT?

**Comment by Hlavtox** [comment]
Can't comment, I don't know anything about it.

If it doesn't do anythinf, remove it. If it does on SOME configurations, keep it.

Lastly, if people are enabling things that they don't know what they do, it's their problem. We don't remove things because some idiot enables them.

**Comment by ShaiMagal** [comment]
@Hlavtox i never saw any positive impact of this settings.

So maybe someone else know what are benefits?

**Comment by jocel1** [comment]
@ShaiMagal I think we don't need it anymore and we can safely remove it as it usually makes more harm than good, and the original use case (which was for prestashop internally) doesn't exist anymore

**Comment by ShaiMagal** [comment]
Can we remove this ? @PrestaShop/committers

**Comment by kpodemski** [comment]
Hello @ShaiMagal 

I would like to point out again that if you want to _discuss_ something, then we use "Discussions", not "Issues" :D 

I've never used it, but based on Jocelyn's insights, I think it is safe to remove it.

**Reply by ShaiMagal** [reply]
@kpodemski I apologize, I didn't know about "Discussions" section :-D Now I know :-D

**Reply by kpodemski** [reply]
@ShaiMagal would you like to remove this setting for the upcoming v9?

**Comment by PululuK** [comment]
Hi @jocel1 , I had recently deploy ps9 with k8s with multiple replicas, I faced some cache issues (the front borke randonly), after have activated this feature it work fine. So I think, we need to keep it for the moment.

**Reply by jocel1** [reply]
Hi @PululuK I agree it fixes this kind of issues, but performances are usually not that good

**Reply by PululuK** [reply]
Yes @jocel1 I agree, that was'nt enought, I did some addicionals improvement to make it work as a wanted (One of was add redis cache), because local sessions storage was a nother point blocking my configuration. Anyway, thank you  ! So I think it is not a good ideia to remove this feature until we found a better solution

**Reply by jocel1** [reply]
Well, my point of view is : it allows Prestashop to work, but with a bad implementation which hurts the performances at the end. In the case of a k8s or any horizontal scaling implementation, we usually care a lot about the performances, so is this really a good idea to keep this solution which prevent from thinking about alternative solutions which really address the issue :) 

BTW "multi-front optimizations" should definitely be renamed to something like "Horizontal scaling compatibility" if we want to keep the option anyway.

**Reply by PululuK** [reply]
Thanks for your feedback !
